### PR TITLE
type[compat]: add html attributes types #4099

### PR DESCRIPTION
**Pull request by samsam-ahmadi**
This PR is for: #4099

### CODE REVIEW

File: compat/src/index.d.ts [hunk 1]
```diff
@@ -68,12 +68,858 @@ declare namespace React {
...
+
+	// this list is "complete" in that it contains every SVG attribute
+	// that React supports, but the types can be improved.
+	// Full list here: https://facebook.github.io/react/docs/dom-elements.html
+	//
+	// The three broad type categories are (in order of restrictiveness):
+	//   - "number | string"
+	//   - "string"
+	//   - union of string literals
+	interface SVGAttributes<T> extends AriaAttributes, DOMAttributes<T> {
```
Comment (rschristian) [inline compat/src/index.d.ts:559]: Any reason why we're not simply reexporting Preact's types here?

https://github.com/preactjs/preact/blob/7cd48335e709f0b32d83070ae320f2010943c310/src/jsx.d.ts#L79

File: compat/src/index.d.ts [hunk 2]
```diff
@@ -68,12 +68,858 @@ declare namespace React {
...
+		nodeintegration?: boolean | undefined;
+		partition?: string | undefined;
+		plugins?: boolean | undefined;
+		preload?: string | undefined;
+		src?: string | undefined;
+		useragent?: string | undefined;
+		webpreferences?: string | undefined;
+	}
+
+	interface ReactSVG {
```
Comment (rschristian) [inline compat/src/index.d.ts:865]: Same here, looks like we're duplicating `InstrinsicElements` definitions?

Comment (samsam-ahmadi) [reply compat/src/index.d.ts:865]: @rschristian Yes we can do this.
I just copied them from react type.

File: compat/src/index.d.ts (hunk 1 above)
Comment (rschristian) [reply compat/src/index.d.ts:559]: We want to avoid duplication as much as possible, copying huge segments from React's types is probably undesirable

Comment (samsam-ahmadi) [reply compat/src/index.d.ts]: Thanks for reviewing and comments. 
I updated the PR and removed the duplicated code and checked others.

Comment (rschristian) [reply compat/src/index.d.ts]: Sorry, I wasn't saying that they should be removed outright, but that we should be re-exporting core's definitions where at all possible.

Most of these are on the JSX namespace in core and therefore do need to be reexported if users want to import the type directly.

Comment (samsam-ahmadi) [reply compat/src/index.d.ts]: Hmm, at first i thought this types are from `IntrinsicElements`. but they not. These types generated by react. 

For example we have `HTMLButtonElement` that comes from mdn, but here we have a new interface of `ButtonHTMLAttributes`

So i don't know is it possible to generate all them by a simple code?

File: src/jsx.d.ts [hunk 3]
```diff
@@ -2021,7 +2021,69 @@ export namespace JSXInternal {
 			| SignalLike<number | string | undefined>;
 	}
 
-	export interface IntrinsicElements {
+	export interface SVGElements {
```
Comment (rschristian) [inline src/jsx.d.ts:2024]:
```suggestion
	export interface SVGInstrinsicElements {
```

or maybe `IntrinsicSVGElements`

File: test-utils/src/index.js [hunk 4]
```diff
@@ -35,14 +35,17 @@ export function act(cb) {
...
-				})
+				return result.then(
+					() => {
+						--actDepth;
+					},
+					e => {
+						--actDepth;
+						throw e;
+					}
+				);
```
Comment (rschristian) [inline test-utils/src/index.js:46]: Should be reverted, keep the diff focused

File: test-utils/test/shared/act.test.js [hunk 5]
```diff
@@ -389,7 +389,7 @@ describe('act', () => {
 			act(() => {
 				tryRenderBroken();
 			});
-		}
```
Comment (rschristian) [inline test-utils/test/shared/act.test.js:392]: Same with the changes here

**Review by rschristian — APPROVED** [review]
Looks very useful, thanks for doing this! We really appreciate it.

Might be nice to move the majority of these into core in the future, as I can see even non-compat users maybe wanting these, but that's an idea for another day.

**Comment by samsam-ahmadi** [comment]
Thanks @rschristian, 
I updated the branch and it's ready to merge.

**Comment by samsam-ahmadi** [comment]
Hi @rschristian,  Any update for this one? How we can get the approval to can merge it?

**Comment by baltarifcan** [comment]
Will this be merged anytime soon?

**Review by rschristian — APPROVED** [review]
I'd like to apologize for this PR, evidently I missed a couple of notifications and it then fell off my radar entirely -- I'm really sorry.

We certainly appreciate any & all contributions, and the time contributors invest into them.

Thanks again.

**Pull request merged**